### ISSUE TITLE
Questions about VAE Training/Fintuning Best Practices

### ISSUE THREAD

**Issue by KokeCacao** [lavinal712/AutoencoderKL#6]
Thank you for providing such great script.

I am new to VAE training. I see two config files `autoencoder_kl_32x32x4.yaml` and `vae-ft-mse-840000-ema-pruned.yaml`. Here are my guesses:
- `autoencoder_kl_32x32x4.yaml` is used for pretrain and `vae-ft-mse-840000-ema-pruned.yaml` is used for finetune
- The reason to modify `perceptual_weight: 0.1` is that perceptual loss isn't always correct for certain style / visual domains. So we use more `pixel_loss` instead.
- I don't understand why you disable `disc_weight` (ie. discriminator loss) for fine-tuning

May I ask whether my guesses are correct?

Edits: maybe it has to do with the following found on https://huggingface.co/stabilityai/sd-vae-ft-mse-original. Do you have any suggestions on config training SDXL VAE?
```
The first, ft-EMA, was resumed from the original checkpoint, trained for 313198 steps and uses EMA weights. It uses the same loss configuration as the original checkpoint (L1 + LPIPS). The second, ft-MSE, was resumed from ft-EMA and uses EMA weights and was trained for another 280k steps using a different loss, with more emphasis on MSE reconstruction (MSE + 0.1 * LPIPS).
```

另外大佬你的博客上不去了.

**Comment by lavinal712** [comment]
> autoencoder_kl_32x32x4.yaml is used for pretrain and vae-ft-mse-840000-ema-pruned.yaml is used for finetune
Yes

> The reason to modify perceptual_weight: 0.1 is that perceptual loss isn't always correct for certain style / visual domains. So we use more pixel_loss instead.
Refer to https://huggingface.co/stabilityai/sd-vae-ft-mse-original

> I don't understand why you disable disc_weight (ie. discriminator loss) for fine-tuning
The currently provided VAE does not release the discriminator weights. Training with an initialized discriminator may affect model performance, so I have set the disc_weight to 0. You're absolutely right that the discriminator can improve model performance during fine-tuning. Therefore, if you have a well-trained discriminator, you can enable it.

> 另外大佬你的博客上不去了.
USTC has removed personal homepages.
